server/task_check_scheduler.go

```diff
@@ -333,6 +333,24 @@ func (s *TaskCheckScheduler) ScheduleCheckIfNeeded(ctx context.Context, task *ap
 			}
 		}
 
+		if s.server.feature(api.FeatureSchemaReviewPolicy) {
```

Comment:
If we do the refactor in the followup PR, then the change here needs to be disabled. Otherwise, every schema change task will trigger the check.